src/health_monitor.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{node::sync_progress::NodeInfo, BlockfrostError, NodePool};
+use std::sync::Arc;
+use tokio::sync::{Notify, RwLock};
+use tokio::time::{self, Duration};
+
+#[derive(Clone)]
+pub struct HealthStatus {
+    pub healthy: bool,
+    pub errors: Vec<BlockfrostError>,
+    pub node_info: Option<NodeInfo>,
+}
+
+#[derive(Clone)]
+pub struct HealthMonitor {
+    inner: Arc<RwLock<HealthStatus>>,
+}
+
+impl HealthMonitor {
+    pub async fn current_status(&self) -> HealthStatus {
+        self.inner.read().await.clone()
+    }
+}
+
+pub async fn spawn(node: NodePool) -> HealthMonitor {
+    // This initial state is never seen:
+    let state_ = Arc::new(RwLock::new(HealthStatus {
+        healthy: false,
+        errors: vec![],
+        node_info: None,
+    }));
+    let state = state_.clone();
+
+    let state_update_ = Arc::new(Notify::new());
+    let state_update = state_update_.clone();
+
+    tokio::spawn(async move {
+        loop {
```

Comment:
in the future, once/if we have other background tasks in some numbers, it will make more sense to have a scheduler that will schedule tasks using only one manager thread. 

For the the need, this is the optimal solution :clap: